hw3/03-form.js

```diff
@@ -0,0 +1,32 @@
+const http = require('http');
+const fs = require('fs');
+const port = process.env.PORT || 5001;
+
+const server = http.createServer((req, res) => {
+    let url = new URL(req.url, `http://${req.headers.host}`);
+    res.setHeader('Content-Type', 'text/html');
+
+    if (req.url === '/') {
+        res.end(fs.readFileSync(__dirname + '/03-form.html'));
+    } else if (url.pathname === '/submitForm') {
+        const parameters = url.searchParams;
+        let text = '';
+        parameters.forEach((value, key) => {
+            if (key == 'Newsletter') {
+                value == 'true'
```

Comment:
The ternary statement excels in short expressions and is used to return one value or another.

Conditionally executing one of two different statements is really the purpose of the if-else statements.

Here you've evaluated one of two different expressions, each wrapped in parentheses, that each return the value returned by  the assignment operator. This value is then not stored anywhere.

```javascript
      if (key == "Newsletter") {
        let test =
          value == "true"
            ? (value = "Yes, I would like to join the newsletter")
            : (value = "No, thank you.");
        console.log("test: ", test);
      }
//the value is still returned from the ternary
```

It is best used as a short-hand for conditionally returning one value or another. People would not expect to see a ternary operator who's value was not captured or that performed side effects that update program state within its arms

Reply:
It is generally advised against in JS to assign into function arguments.
consider:
```javascript
let newValue =
  value === "true"
    ? "Yes, I would like to join the newsletter"
    : "No, thank you.";
```
or:
```javascript
if (key === "Newsletter") {
  let newValue;
  if (value === "true") {
    newValue = "Yes, I would like to join the newsletter";
  } else {
    newValue = "No, thank you.";
  }
}
```